src/hooks/cgroups/pbs_cgroups.PY

```diff
@@ -0,0 +1,3657 @@
+                        pbs.logmsg(pbs.EVENT_DEBUG2, "%s: val = %s (%s)" %
+                                   (caller_name(), val, type(val)))
+                        if key not in host_resc_avail:
+                            host_resc_avail[key] = initialize_resource(val)
+                        else:
+                            if not host_resc_avail[key]:
+                                host_resc_avail[key] = initialize_resource(val)
+                        host_resc_avail[key] += val
+        pbs.logmsg(pbs.EVENT_DEBUG3, "%s: vnode list: %s" %
+                   (caller_name(), vnode_list))
+        if vnodes:
+            pbs.logmsg(pbs.EVENT_DEBUG3, "%s: vnode_resc_avail: %s" %
+                       (caller_name(), vnode_resc_avail))
+        pbs.logmsg(pbs.EVENT_DEBUG3, "%s: host_resc_avail: %s" %
+                   (caller_name(), host_resc_avail))
+        return True
+
+
+#
+# CLASS CgroupUtils
+#
+class CgroupUtils:
+
+    def __init__(self, hostname, vnode, cfg=None, subsystems=None,
+                 paths=None, vntype=None, assigned_resources=None):
+        self.hostname = hostname
+        self.vnode = vnode
+        # __check_os will raise an exception if cgroups are not present
+        self.__check_os()
+        # Read in the config file
+        if cfg is not None:
+            self.cfg = cfg
+        else:
+            self.cfg = self.__parse_config_file()
+        # Collect the cgroup mount points
+        if paths is not None:
+            self.paths = paths
+        else:
+            self.paths = self.__get_paths()
+        # Raise an error if nothing is mounted
+        if not self.paths:
+            pbs.logmsg(pbs.EVENT_DEBUG2, "%s: No cgroups mounted" %
+                       (caller_name()))
+            raise CgroupProcessingError("No CPUs avaialble in cgroup.")
+        # Define the local vnode type
+        if vntype is not None:
+            self.vntype = vntype
+        else:
+            self.vntype = self.__get_vnode_type()
+        # Determine which subsystems we care about
+        if subsystems is not None:
+            self.subsystems = subsystems
+        else:
+            self.subsystems = self.__target_subsystems()
+        # Return now if nothing is enabled
+        if not self.subsystems:
+            pbs.logmsg(pbs.EVENT_DEBUG2, "%s: No cgroups enabled" %
+                       (caller_name()))
+            self.assigned_resources = {}
+            return
+        # location to store information for the different hook events
+        self.hook_storage_dir = os.path.join(pbs_home, 'mom_priv',
+                                             'hooks', 'hook_data')
+        self.host_job_env_dir = os.path.join(pbs_home, 'aux')
+        self.host_job_env_filename = os.path.join(self.host_job_env_dir,
+                                                  '%s.env')
+        # information for offlining nodes
+        self.offline_file = os.path.join(pbs_home, 'mom_priv', 'hooks',
+               
```

Comment:
This might have been one of those comments that went to the wrong place.  I don't mind variable names like 'offline_file'.  Using 'file' by itself is a problem.